compiler/ilgen/IlBuilder.cpp

```diff
@@ -1419,6 +1419,49 @@ IlBuilder::Call(const char *functionName, int32_t numArgs, TR::IlValue ** argVal
    return NULL;
    }
 
+TR::IlValue *
+IlBuilder::AtomicIntegerAdd(TR::IlValue * object, TR::IlValue * offset, TR::IlValue * value)
+   {
+   TraceIL("IlBuilder[ %p ]::AtomicIntegerAdd (%d, %d, %d)\n", this, object->getCPIndex(), offset->getCPIndex(), value->getCPIndex());
+   appendBlock();
+
+   TR::DataType returnType = TR::Int32;
+   
+   TR::SymbolReference *methodSymRef = symRefTab()->findOrCreateCodeGenInlinedHelper(TR::SymbolReferenceTable::atomicAdd32BitSymbol); //lock add
```

Comment:
There is an implicit assumption here of the type - shouldn't there be some kind of type check on the values passed in to make sure they are of the correct type and to determine the kind of atomic operation required? On 64bit platforms we can support 64bit atomic operations for example. At the very least some asserts and code documentation is probably warranted.